From 80c611d8d54c76e7cb239be6cbc8b0e3adacd692 Mon Sep 17 00:00:00 2001
From: Taku Imaizumi <taku.imaizumi.bx@bp.renesas.com>
Date: Fri, 24 Jul 2015 21:07:11 +0900
Subject: [PATCH 7/7] Change to be able to enter the environment variable(CC)

---
 flags.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/flags.mk b/flags.mk
index 92e1853..3c8b728 100644
--- a/flags.mk
+++ b/flags.mk
@@ -3,7 +3,7 @@
 #########################################################################
 
 CROSS_COMPILE   ?= arm-linux-gnueabihf-
-CC              := $(CROSS_COMPILE)gcc
+CC              ?= $(CROSS_COMPILE)gcc
 
 CFLAGS          := -Wall -Wbad-function-cast -Wcast-align \
 		   -Werror-implicit-function-declaration -Wextra \
-- 
1.9.5.msysgit.1

