From 2035e9a30a235204b3d96ec369ad601dfd93579d Mon Sep 17 00:00:00 2001
From: Jun Miyauchi <jun.miyauchi.sr@hitachi.com>
Date: Fri, 22 Dec 2017 09:13:57 +0900
Subject: [PATCH] Add LDFLAGS option

---
 libteec/Makefile        | 2 +-
 tee-supplicant/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libteec/Makefile b/libteec/Makefile
index 568a2cd..a9cca19 100644
--- a/libteec/Makefile
+++ b/libteec/Makefile
@@ -45,7 +45,7 @@ libteec: $(TEEC_LIBRARY)
 
 $(TEEC_LIBRARY): $(TEEC_OBJS)
 	@echo "  LINK    $@"
-	$(VPREFIX)$(CC) -shared -Wl,-soname,$(LIB_MAJ_MIN) $(TEEC_LFLAGS) -o $@ $+
+	$(VPREFIX)$(CC) $(LDFLAGS) -shared -Wl,-soname,$(LIB_MAJ_MIN) $(TEEC_LFLAGS) -o $@ $+
 	@echo ""
 
 $(TEEC_OBJ_DIR)/%.o: ${TEEC_SRC_DIR}/%.c
diff --git a/tee-supplicant/Makefile b/tee-supplicant/Makefile
index 737d39a..98e7408 100644
--- a/tee-supplicant/Makefile
+++ b/tee-supplicant/Makefile
@@ -69,7 +69,7 @@ tee-supplicant: $(TEES_FILE)
 
 $(TEES_FILE): $(TEES_OBJS)
 	@echo "  LINK    $@"
-	$(VPREFIX)$(CC) -o $@ $+ $(TEES_LDFLAGS)
+	$(VPREFIX)$(CC) $(LDFLAGS) -o $@ $+ $(TEES_LDFLAGS)
 	@echo ""
 
 $(TEES_OBJ_DIR)/%.o: $(TEES_SRC_DIR)/%.c
-- 
2.14.1.windows.1

