From 4c13d95d07f5eace4830876f21c9407600be11fd Mon Sep 17 00:00:00 2001
From: Andreas Shimokawa <andi@directfb.org>
Date: Tue, 21 May 2013 17:05:23 +0200
Subject: [PATCH 3/3] drmkms: replace DLCONF_ with CLRCF_ where appropriate.

---
 systems/drmkms/drmkms_layer.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/systems/drmkms/drmkms_layer.c b/systems/drmkms/drmkms_layer.c
index 65ecdfa..b55db62 100644
--- a/systems/drmkms/drmkms_layer.c
+++ b/systems/drmkms/drmkms_layer.c
@@ -101,7 +101,7 @@ drmkmsTestRegion( CoreLayer                  *layer,
 
           D_DEBUG_AT( DRMKMS_Layer, "    -> rejecting layer that is smaller than the screen (drm/kms limitation)\n" );
           if (ret_failed)
-               *ret_failed = DLCONF_WIDTH | DLCONF_HEIGHT;
+               *ret_failed = CLRCF_WIDTH | CLRCF_HEIGHT;
 
           return DFB_UNSUPPORTED;
      }
@@ -391,11 +391,11 @@ drmkmsPlaneTestRegion( CoreLayer                  *layer,
 {
      DRMKMSLayerData *data  = layer_data;
 
-     CoreLayerRegionConfigFlags failed = DLCONF_NONE;
+     CoreLayerRegionConfigFlags failed = CLRCF_NONE;
 
      if (((config->options & DLOP_OPACITY     ) && !data->alpha_propid   ) ||
          ((config->options & DLOP_SRC_COLORKEY) && !data->colorkey_propid))
-          failed |= DLCONF_OPTIONS;
+          failed |= CLRCF_OPTIONS;
 
      if (ret_failed)
           *ret_failed = failed;
-- 
1.8.1.2

