From eab8a7456236d0894faf3925ec83662843427306 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Tue, 24 Apr 2012 16:51:32 +0900
Subject: [PATCH 14/97] ext/dfbvideosink: enable scaling if libshvio blitting
 used

---
 ext/directfb/dfbvideosink.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 95b3438..5feab23 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1682,7 +1682,11 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
     res = surface->GetSize (surface, &dst.w, &dst.h);
 
     /* Center / Clip */
+#if defined(HAVE_SHVIO)
+    gst_video_sink_center_rect (src, dst, &result, TRUE);
+#else
     gst_video_sink_center_rect (src, dst, &result, FALSE);
+#endif
 
     res =
         surface->GetSubSurface (surface, (DFBRectangle *) (void *) &result,
-- 
1.7.10.4

