From caa800138874f676602b9ac881843db80816a361 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 26 Jul 2012 15:17:17 +0900
Subject: [PATCH 63/97] dfb-example: clear another surface of double buffer

---
 ext/directfb/dfb-example.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/directfb/dfb-example.c b/ext/directfb/dfb-example.c
index b361631..e805911 100644
--- a/ext/directfb/dfb-example.c
+++ b/ext/directfb/dfb-example.c
@@ -211,6 +211,8 @@ main (int argc, char *argv[])
   }
 
   primary->Clear (primary, 0x00, 0x00, 0x00, 0xFF);
+  primary->Flip (primary, NULL, DSFLIP_NONE);
+  primary->Clear (primary, 0x00, 0x00, 0x00, 0xFF);
 
   /* Let's play ! */
   gst_element_set_state (pipeline, GST_STATE_PLAYING);
-- 
1.7.10.4

