From 6da019672c1fc19f41c58aeac65d725e42a67fc1 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 26 Jul 2012 16:53:29 +0900
Subject: [PATCH 65/97] dfb-example: fix lacks of finalizing processing

---
 ext/directfb/dfb-example.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/directfb/dfb-example.c b/ext/directfb/dfb-example.c
index bfecac7..0dbbe92 100644
--- a/ext/directfb/dfb-example.c
+++ b/ext/directfb/dfb-example.c
@@ -225,8 +225,11 @@ main (int argc, char *argv[])
 
   /* Release DirectFB context and surface */
   primary->Release (primary);
+  layer->Release (layer);
   dfb->Release (dfb);
 
+  gst_deinit ();
+
   for (i = 1; i < tmp_argc; i++) {
     free (tmp_argv[i]);
   }
-- 
1.7.10.4

