From aeab4f72cce152bbb01ccd70fac76ae934947ce8 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 15 Oct 2012 10:34:05 +0900
Subject: [PATCH 71/97] ext/dfbvideosink: reflect tile boundary offset on
 MExxBSIZE

---
 ext/directfb/dfbvideosink.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index e03f494..fe7ab07 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1649,7 +1649,8 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
       0xa0000701);
   meram_write_icb (dfbvideosink->meram, dfbvideosink->icby[SRC], MExxSSARB, 0);
   meram_write_icb (dfbvideosink->meram, dfbvideosink->icby[SRC], MExxBSIZE,
-      (((sliceheight - 1) & 0x1fff) << 16) | ((stride - 1) & 0x7fff));
+      (((sliceheight + dfbvideosink->tile_boundary_y_offset -
+                  1) & 0x1fff) << 16) | ((stride - 1) & 0x7fff));
   sbsize = stride & 0xfff0;
   switch (tilewidth) {
     case 16:
@@ -1691,7 +1692,8 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
     meram_write_icb (dfbvideosink->meram, dfbvideosink->icbc[SRC], MExxCTRL,
         0xa0200701);
     meram_write_icb (dfbvideosink->meram, dfbvideosink->icbc[SRC], MExxBSIZE,
-        (((sliceheight / 2 - 1) & 0x1fff) << 16) | ((stride - 1) & 0x7fff));
+        (((sliceheight / 2 + dfbvideosink->tile_boundary_c_offset -
+                    1) & 0x1fff) << 16) | ((stride - 1) & 0x7fff));
     meram_write_icb (dfbvideosink->meram, dfbvideosink->icbc[SRC], MExxSBSIZE,
         sbsize);
     meram_write_icb (dfbvideosink->meram, dfbvideosink->icbc[SRC], MExxSSARB,
-- 
1.7.10.4

