From f1bf5380d0f227e689d32f58af89a69268fcd758 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 31 Oct 2012 18:51:36 +0900
Subject: [PATCH 91/97] dfb-video-example: fix incorrect log messages

---
 ext/directfb/dfb-video-example.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/directfb/dfb-video-example.c b/ext/directfb/dfb-video-example.c
index 07b4090..c8bee91 100644
--- a/ext/directfb/dfb-video-example.c
+++ b/ext/directfb/dfb-video-example.c
@@ -157,10 +157,10 @@ create_audio_pipeline (GstPad * pad)
     gst_structure_get_int (structure, "layer", &layer);
     /* This test to determine is based on "http://gstreamer.freedesktop.org/data/doc/gstreamer/head/pwg/html/section-types-definitions.html". */
     if (ver == 4 || ver == 2) { /* AAC */
-      printf ("codec type video/mpeg mpegversion=%d\n", ver);
+      printf ("codec type audio/mpeg mpegversion=%d\n", ver);
       decoder = gst_element_factory_make ("faad", NULL);
     } else if (ver == 1 && layer == 3) {        /* MP3 */
-      printf ("codec type video/mpeg mpegversion=1\n");
+      printf ("codec type audio/mpeg mpegversion=%d\n", ver);
       decoder = gst_element_factory_make ("mad", NULL);
     } else {
       printf ("unsupported format\n");
-- 
1.7.10.4

