diff --git a/gio/Makefile.am b/gio/Makefile.am
index b0ae4de..75fd3e3 100644
--- a/gio/Makefile.am
+++ b/gio/Makefile.am
@@ -647,6 +647,8 @@ bin_PROGRAMS = gio-querymodules glib-compile-schemas glib-compile-resources gset
 glib_compile_resources_LDADD = \
 	$(top_builddir)/glib/libglib-2.0.la \
 	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la       \
+	$(top_builddir)/gmodule/libgmodule-2.0.la       \
 	libgio-2.0.la
 
 glib_compile_resources_SOURCES = \
@@ -661,6 +663,7 @@ gio_querymodules_LDADD	 = \
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	libgio-2.0.la					\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
 	$(NULL)
 
 gconstructor_as_data.h: $(top_srcdir)/glib/gconstructor.h data-to-c.pl
@@ -677,6 +680,8 @@ glib_compile_schemas_SOURCES = \
 gsettings_LDADD = \
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
 	libgio-2.0.la
 gsettings_SOURCES = gsettings-tool.c
 
@@ -690,7 +695,8 @@ bin_PROGRAMS += gdbus
 gdbus_SOURCES = gdbus-tool.c
 gdbus_LDADD = libgio-2.0.la \
 	$(top_builddir)/glib/libglib-2.0.la 		\
-	$(top_builddir)/gobject/libgobject-2.0.la
+	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la
 
 completiondir = $(sysconfdir)/bash_completion.d
 completion_DATA = \
@@ -708,6 +714,8 @@ gresource_CPPFLAGS = $(LIBELF_CFLAGS) $(AM_CPPFLAGS)
 gresource_LDADD = libgio-2.0.la \
 	$(top_builddir)/glib/libglib-2.0.la 		\
 	$(top_builddir)/gobject/libgobject-2.0.la \
+	$(top_builddir)/gthread/libgthread-2.0.la       \
+	$(top_builddir)/gmodule/libgmodule-2.0.la       \
 	$(LIBELF_LIBS)
 
 # ------------------------------------------------------------------------
diff --git a/gio/tests/Makefile.am b/gio/tests/Makefile.am
index 0d45e71..c293bf3 100644
--- a/gio/tests/Makefile.am
+++ b/gio/tests/Makefile.am
@@ -23,7 +23,8 @@ progs_ldadd     = 					\
 	$(top_builddir)/gthread/libgthread-2.0.la	\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	$(top_builddir)/gmodule/libgmodule-2.0.la	\
-	$(top_builddir)/gio/libgio-2.0.la
+	$(top_builddir)/gio/libgio-2.0.la               \
+	$(top_builddir)/gmodule/libgmodule-2.0.la
 
 TEST_PROGS +=	 		\
 	io-stream		\
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 8f4c57e..9cb802b 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -77,8 +77,8 @@ unicode_normalize_LDADD = $(libglib)
 unicode_collate_LDADD = $(libglib)
 assert_msg_test_LDADD = $(libglib)
 if ENABLE_TIMELOOP
-timeloop_LDADD = $(libglib)
-timeloop_closure_LDADD = $(libglib) $(libgobject)
+timeloop_LDADD = $(libglib) $(libgthread)
+timeloop_closure_LDADD = $(libglib) $(libgobject) $(libgthread)
 endif
 datetime_LDADD = $(libglib)
 
