From f34904b362c133c04545834aaea4600cbac8bac3 Mon Sep 17 00:00:00 2001
From: Yoshifumi Hosoya <yoshifumi.hosoya.wj@renesas.com>
Date: Thu, 20 Aug 2015 13:16:04 +0900
Subject: [PATCH 26/85] ravb: Add RAVB config for R-Car Gen3

Signed-off-by: Yoshifumi Hosoya <yoshifumi.hosoya.wj@renesas.com>
Signed-off-by: Masaru Nagai <masaru.nagai.vx@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
---
 drivers/net/ethernet/renesas/Kconfig |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/renesas/Kconfig b/drivers/net/ethernet/renesas/Kconfig
index 270c4c9..e5dd283 100644
--- a/drivers/net/ethernet/renesas/Kconfig
+++ b/drivers/net/ethernet/renesas/Kconfig
@@ -4,6 +4,8 @@
 
 config NET_VENDOR_RENESAS
 	bool "Renesas devices"
+	depends on ARCH_RCAR_GEN3 || ARCH_SHMOBILE
+	select RAVB
 	default y
 	---help---
 	  If you have a network (Ethernet) card belonging to this class, say Y.
-- 
1.7.9.5

