diff --git a/xf-rcar.c b/xf-rcar.c
index 20ed545..cfa9f0c 100644
--- a/xf-rcar.c
+++ b/xf-rcar.c
@@ -521,7 +521,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 		return -ENOENT;
 	}
 
-	ipc->regs = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->regs = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->regs)) {		/* PRQA S 0306 2 */
 		res = (int)PTR_ERR(ipc->regs);
 		pr_err("ioremap failed regs: %d", res);
@@ -590,7 +590,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->dtcm1.res = mmio;
 
-	ipc->dtcm1.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->dtcm1.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->dtcm1.kaddr)) {		/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->dtcm1.kaddr);
 		pr_err("ioremap failed dtcm1: %d", res);
@@ -604,7 +604,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->dtcm0.res = mmio;
 
-	ipc->dtcm0.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->dtcm0.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->dtcm0.kaddr)) {			/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->dtcm0.kaddr);
 		pr_err("ioremap failed dtcm0: %d", res);
@@ -618,7 +618,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->itcm0.res = mmio;
 
-	ipc->itcm0.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->itcm0.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->itcm0.kaddr)) {			/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->itcm0.kaddr);
 		pr_err("ioremap failed itcm0: %d", res);
@@ -632,7 +632,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->itcm1.res = mmio;
 
-	ipc->itcm1.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->itcm1.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->itcm1.kaddr)) {			/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->itcm1.kaddr);
 		pr_err("ioremap failed itcm1: %d", res);
@@ -647,7 +647,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->dtcm1.res = mmio;
 
-	ipc->dtcm1.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->dtcm1.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->dtcm1.kaddr)) {		/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->dtcm1.kaddr);
 		pr_err("ioremap failed dtcm1: %d", res);
@@ -661,7 +661,7 @@ static int __init lx_adsp_probe(struct platform_device *pdev)		/* PRQA S 3450 */
 	}
 	ipc->itcm0.res = mmio;
 
-	ipc->itcm0.kaddr = devm_ioremap_nocache(&pdev->dev, mmio->start, resource_size(mmio));
+	ipc->itcm0.kaddr = devm_ioremap(&pdev->dev, mmio->start, resource_size(mmio));
 	if (IS_ERR(ipc->itcm0.kaddr)) {			/* PRQA S 0306 2*/
 		res = (int)PTR_ERR(ipc->itcm0.kaddr);
 		pr_err("ioremap failed itcm0: %d", res);
