From 264fcff0207314ad61286823b00a57da427dd345 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 23 Jun 2016 15:13:29 +0300
Subject: [PATCH] workaround gcc optimisation

---
 drivers/mmc/sh_sdhi.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/mmc/sh_sdhi.c b/drivers/mmc/sh_sdhi.c
index cc62c89..58e5250 100644
--- a/drivers/mmc/sh_sdhi.c
+++ b/drivers/mmc/sh_sdhi.c
@@ -401,10 +401,12 @@ static void sh_sdhi_get_response(struct sh_sdhi_host *host, struct mmc_cmd *cmd)
 	unsigned short resp[8];
 	unsigned long *p1, *p2;
 
+	resp[0] = sh_sdhi_readw(host, SDHI_RSP00);
+	resp[1] = sh_sdhi_readw(host, SDHI_RSP01);
+	p1 = ((unsigned long *)resp);
+
 	if (cmd->resp_type & MMC_RSP_136) {
 		cnt = 4;
-		resp[0] = sh_sdhi_readw(host, SDHI_RSP00);
-		resp[1] = sh_sdhi_readw(host, SDHI_RSP01);
 		resp[2] = sh_sdhi_readw(host, SDHI_RSP02);
 		resp[3] = sh_sdhi_readw(host, SDHI_RSP03);
 		resp[4] = sh_sdhi_readw(host, SDHI_RSP04);
@@ -422,11 +424,6 @@ static void sh_sdhi_get_response(struct sh_sdhi_host *host, struct mmc_cmd *cmd)
 		/* SDHI REGISTER SPECIFICATION */
 		p1 = ((unsigned long *)resp) + 3;
 
-	} else {
-		resp[0] = sh_sdhi_readw(host, SDHI_RSP00);
-		resp[1] = sh_sdhi_readw(host, SDHI_RSP01);
-
-		p1 = ((unsigned long *)resp);
 	}
 
 	p2 = (unsigned long *)cmd->response;
-- 
1.9.1

