From 79137df3ddc365922d4ab15d5b44bf4800812062 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Fri, 15 Jul 2016 10:41:12 +0300
Subject: [PATCH] fix da9063 address

---
 include/configs/skrzg1m.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/skrzg1m.h b/include/configs/skrzg1m.h
index 59b14e9..c8fd510 100644
--- a/include/configs/skrzg1m.h
+++ b/include/configs/skrzg1m.h
@@ -82,7 +82,7 @@
 #define CONFIG_SH_I2C_DATA_LOW		5
 #define CONFIG_SH_I2C_CLOCK		10000000
 
-#define CONFIG_SYS_I2C_POWERIC_ADDR	0x58 /* da9063 */
+#define CONFIG_SYS_I2C_POWERIC_ADDR	0x5A /* da9063 */
 
 /* USB */
 #define CONFIG_USB_EHCI
-- 
1.9.1

