From d457020ec25198a826acf634420829a537af8906 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Mon, 18 Jul 2016 20:16:52 +0300
Subject: [PATCH] reduce speed on i2c

---
 include/configs/silk.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/silk.h b/include/configs/silk.h
index cd839aa..f9b8341 100644
--- a/include/configs/silk.h
+++ b/include/configs/silk.h
@@ -76,7 +76,7 @@
 #define CONFIG_SYS_I2C_SLAVE		0x7F
 #define CONFIG_SYS_I2C_SH_NUM_CONTROLLERS	3
 #define CONFIG_SYS_I2C_SH_SPEED0	400000
-#define CONFIG_SYS_I2C_SH_SPEED1	400000
+#define CONFIG_SYS_I2C_SH_SPEED1	100000
 #define CONFIG_SYS_I2C_SH_SPEED2	400000
 #define CONFIG_SH_I2C_DATA_HIGH		4
 #define CONFIG_SH_I2C_DATA_LOW		5
-- 
2.7.4

