From 7081a5120a5dcb6b1af1eb0f748c56cbf5e4b22a Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Mon, 11 Apr 2016 12:24:57 +0300
Subject: [PATCH] rename clks

---
 vspm/drv/vspm_sub.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vspm/drv/vspm_sub.c b/vspm/drv/vspm_sub.c
index 54818a2..01aad20 100755
--- a/vspm/drv/vspm_sub.c
+++ b/vspm/drv/vspm_sub.c
@@ -116,7 +116,7 @@ int vspm_init(struct vspm_privdata *priv)
 	pm_runtime_get_sync(&pdev->dev);
 
 	/* enable clock */
-	pdrv->tddmac_clk = clk_get(NULL, "tddmac");
+	pdrv->tddmac_clk = clk_get(NULL, "2ddmac");
 	if (IS_ERR(pdrv->tddmac_clk)) {
 		APRINT("failed to get 2DDMAC clock\n");
 		ercd = -EFAULT;
@@ -130,7 +130,7 @@ int vspm_init(struct vspm_privdata *priv)
 		goto err_exit2;
 	}
 
-	pdrv->vsps_clk = clk_get(NULL, "vsps");
+	pdrv->vsps_clk = clk_get(NULL, "vsp1-sy");
 	if (IS_ERR(pdrv->vsps_clk)) {
 		APRINT("failed to get VSPS clock\n");
 		ercd = -EFAULT;
-- 
1.9.1

