From 6d7f09e4374f4951d3de3789b66bc3a0108fdde4 Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Wed, 17 Oct 2012 10:59:35 +0100
Subject: [PATCH 72/76] arm: shmobile: r8a7779: Add SGX clock

---
 arch/arm/mach-shmobile/clock-r8a7779.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-shmobile/clock-r8a7779.c b/arch/arm/mach-shmobile/clock-r8a7779.c
index 8398b70..7303b73 100644
--- a/arch/arm/mach-shmobile/clock-r8a7779.c
+++ b/arch/arm/mach-shmobile/clock-r8a7779.c
@@ -87,7 +87,8 @@ static struct clk div4_clks[DIV4_NR] = {
 };
 
 enum { MSTP323, MSTP322, MSTP321, MSTP320,
-	MSTP120, MSTP114, MSTP110, MSTP109, MSTP108, MSTP103, MSTP101, MSTP100,
+	MSTP120, MSTP114, MSTP110, MSTP109,
+	MSTP108, MSTP105, MSTP103, MSTP101, MSTP100,
 	MSTP030, MSTP029, MSTP028, MSTP027, MSTP026, MSTP025, MSTP024, MSTP023,
 	MSTP022, MSTP021, MSTP016, MSTP015, MSTP014, MSTP012, MSTP011, MSTP008,
 	MSTP007,
@@ -103,6 +104,7 @@ static struct clk mstp_clks[MSTP_NR] = {
 	[MSTP110] = SH_CLK_MSTP32(&div4_clks[DIV4_S], MSTPCR1, 10, 0), /* VIN0 */
 	[MSTP109] = SH_CLK_MSTP32(&div4_clks[DIV4_S], MSTPCR1,  9, 0), /* VIN1 */
 	[MSTP108] = SH_CLK_MSTP32(&div4_clks[DIV4_S], MSTPCR1,  8, 0), /* VIN2 */
+	[MSTP105] = SH_CLK_MSTP32(&div4_clks[DIV4_S], MSTPCR1,  5, 0), /* SGX */
 	[MSTP103] = SH_CLK_MSTP32(&div4_clks[DIV4_S], MSTPCR1,  3, 0), /* DU */
 	[MSTP101] = SH_CLK_MSTP32(&div4_clks[DIV4_P], MSTPCR1,  1, 0), /* USB2 */
 	[MSTP100] = SH_CLK_MSTP32(&div4_clks[DIV4_P], MSTPCR1,  0, 0), /* USB01 */
@@ -188,6 +190,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_DEV_ID("rcar_vin.0", &mstp_clks[MSTP110]), /* VIN0 */
 	CLKDEV_DEV_ID("rcar_vin.1", &mstp_clks[MSTP109]), /* VIN1 */
 	CLKDEV_DEV_ID("rcar_vin.2", &mstp_clks[MSTP108]), /* VIN2 */
+	CLKDEV_CON_ID("sgx", &mstp_clks[MSTP105]), /* SGX */
 	CLKDEV_DEV_ID("rcarfb.0", &mstp_clks[MSTP103]), /* DU0 */
 	CLKDEV_DEV_ID("rcarfb.1", &mstp_clks[MSTP103]), /* DU1 */
 
-- 
1.7.5.4

