diff --git a/xf-ap-msg.h b/xf-ap-msg.h
index 50acc28..1a7b636 100644
--- a/xf-ap-msg.h
+++ b/xf-ap-msg.h
@@ -125,7 +125,7 @@ typedef struct xf_ext_param_msg
     /* ...parameter data (in the format expected by codec) */
     u8                      data[0];
 
-}   __attribute__ ((__packed__)) xf_ext_param_msg_t;
+}   __attribute__ ((__packed__, __aligned__(4))) xf_ext_param_msg_t;
 
 /* ...access macros */
 
