From 1681be6973e5717ac542750b845ca33c3f94dfd5 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 15 Apr 2013 14:58:51 +0900
Subject: [PATCH 03/17] ARM: shmobile: r8a7778: add IPMMU support

---
 arch/arm/mach-shmobile/setup-r8a7778.c |   28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm/mach-shmobile/setup-r8a7778.c b/arch/arm/mach-shmobile/setup-r8a7778.c
index a3dce67..1515f41 100644
--- a/arch/arm/mach-shmobile/setup-r8a7778.c
+++ b/arch/arm/mach-shmobile/setup-r8a7778.c
@@ -33,6 +33,7 @@
 #include <linux/serial_sci.h>
 #include <linux/sh_intc.h>
 #include <linux/sh_timer.h>
+#include <linux/uio_driver.h>
 #include <mach/hardware.h>
 #include <mach/irqs.h>
 #include <mach/r8a7778.h>
@@ -308,6 +309,32 @@ static struct platform_device i2c3_device = {
 	.num_resources	= ARRAY_SIZE(rcar_i2c3_res),
 };
 
+/* IPMMU */
+static struct uio_info ipmmu_platform_data = {
+	.name = "IPMMU",
+	.version = "0",
+	.irq = -1,
+};
+
+static struct resource ipmmu_resources[] = {
+	[0] = {
+		.name   = "IPMMU",
+		.start  = 0xfd051000,
+		.end    = 0xfd0510ff,
+		.flags  = IORESOURCE_MEM,
+	},
+};
+
+static struct platform_device ipmmu_device = {
+	.name           = "uio_pdrv_genirq",
+	.id             = 0,
+	.dev = {
+		.platform_data = &ipmmu_platform_data,
+	},
+	.resource       = ipmmu_resources,
+	.num_resources  = ARRAY_SIZE(ipmmu_resources),
+};
+
 static struct platform_device *r8a7778_early_devices[] __initdata = {
 	&scif0_device,
 	&scif1_device,
@@ -321,6 +348,7 @@ static struct platform_device *r8a7778_early_devices[] __initdata = {
 	&i2c1_device,
 	&i2c2_device,
 	&i2c3_device,
+	&ipmmu_device,
 };
 
 static struct platform_device *r8a7778_late_devices[] __initdata = {
-- 
1.7.9.5

