From bca2175261280a1f573d3f63fad1bc00682cdfed Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 15 Apr 2013 13:50:54 +0900
Subject: [PATCH 07/17] ARM: shmobile: bockw: invoke ICB MERAM workaround

---
 arch/arm/mach-shmobile/board-bockw.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index e063de7..4f9c780 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -426,6 +426,7 @@ static void __init bockw_init(void)
 	 */
 
 	r8a7778_pinmux_init();
+	r8a7778_meram_workaround();
 
 	/* interrupt mask on IRQ0 */
 	val = ioread16(IRQ0MR);
-- 
1.7.9.5

