From 189def67cd4c102a4ab75f8d350bfaceeb107975 Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Tue, 16 Apr 2013 17:13:50 +0900
Subject: [PATCH 16/17] Local: ARM: shmobile: r8a7778: Fix INT num for
 I2C1/2/3

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7778.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7778.c b/arch/arm/mach-shmobile/setup-r8a7778.c
index 0a72e86..ce4dc21 100644
--- a/arch/arm/mach-shmobile/setup-r8a7778.c
+++ b/arch/arm/mach-shmobile/setup-r8a7778.c
@@ -263,7 +263,7 @@ static struct resource rcar_i2c1_res[] = {
 		.end	= 0xffc71fff,
 		.flags	= IORESOURCE_MEM,
 	}, {
-		.start	= gic_spi(76),
+		.start	= gic_spi(78),
 		.flags	= IORESOURCE_IRQ,
 	},
 };
@@ -281,7 +281,7 @@ static struct resource rcar_i2c2_res[] = {
 		.end	= 0xffc72fff,
 		.flags	= IORESOURCE_MEM,
 	}, {
-		.start	= gic_spi(77),
+		.start	= gic_spi(76),
 		.flags	= IORESOURCE_IRQ,
 	},
 };
@@ -299,7 +299,7 @@ static struct resource rcar_i2c3_res[] = {
 		.end	= 0xffc73fff,
 		.flags	= IORESOURCE_MEM,
 	}, {
-		.start	= gic_spi(78),
+		.start	= gic_spi(77),
 		.flags	= IORESOURCE_IRQ,
 	},
 };
-- 
1.7.9.5

