From 1b5a3514a69479a663104e445ee0d41abcce5bdd Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 11 Jun 2013 17:26:14 +0900
Subject: [PATCH 579/675] sh-pfc: r8a7778: add DU pin support

---
 drivers/pinctrl/sh-pfc/pfc-r8a7778.c | 54 ++++++++++++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a7778.c b/drivers/pinctrl/sh-pfc/pfc-r8a7778.c
index 50c066a..fdbe196 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a7778.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a7778.c
@@ -1772,6 +1772,44 @@ VIN_PFC_PINS(vin1_clk,		RCAR_GP_PIN(4, 9));
 VIN_PFC_CLK(vin1_clk,		VI1_CLK);
 VIN_PFC_PINS(vin1_sync,		RCAR_GP_PIN(3, 21),	RCAR_GP_PIN(3, 22));
 VIN_PFC_SYNC(vin1_sync,		VI1_HSYNC,		VI1_VSYNC);
+/* - DU macros ------------------------------------------------------------- */
+#define DU_PFC_PINS(name, args...)		SH_PFC_PINS(name, args)
+#define DU_PFC_DAT8(name, d0, d1, d2, d3, d4, d5, d6, d7)	\
+	SH_PFC_MUX8(name, d0, d1, d2, d3, d4, d5, d6, d7)
+#define DU_PFC_CLK(name, clk0, clk1)		SH_PFC_MUX2(name, clk0, clk1)
+#define DU_PFC_DISP(name, disp)			SH_PFC_MUX1(name, disp)
+#define DU_PFC_SYNC(name, hsync, vsync)	SH_PFC_MUX2(name, hsync, vsync)
+/* - DU0 ------------------------------------------------------------------- */
+DU_PFC_PINS(du0_data8r,		RCAR_GP_PIN(1, 22),	RCAR_GP_PIN(1, 23),
+				RCAR_GP_PIN(1, 24),	RCAR_GP_PIN(1, 25),
+				RCAR_GP_PIN(1, 26),	RCAR_GP_PIN(1, 27),
+				RCAR_GP_PIN(1, 28),	RCAR_GP_PIN(1, 29));
+DU_PFC_DAT8(du0_data8r,		DU0_DR0,		DU0_DR1,
+				DU0_DR2,		DU0_DR3,
+				DU0_DR4,		DU0_DR5,
+				DU0_DR6,		DU0_DR7);
+DU_PFC_PINS(du0_data8g,		RCAR_GP_PIN(1, 30),	RCAR_GP_PIN(1, 31),
+				RCAR_GP_PIN(2, 0),	RCAR_GP_PIN(2, 1),
+				RCAR_GP_PIN(2, 2),	RCAR_GP_PIN(2, 3),
+				RCAR_GP_PIN(2, 4),	RCAR_GP_PIN(2, 5));
+DU_PFC_DAT8(du0_data8g,		DU0_DG0,		DU0_DG1,
+				DU0_DG2,		DU0_DG3,
+				DU0_DG4,		DU0_DG5,
+				DU0_DG6,		DU0_DG7);
+DU_PFC_PINS(du0_data8b,		RCAR_GP_PIN(2, 6),	RCAR_GP_PIN(2, 7),
+				RCAR_GP_PIN(2, 8),	RCAR_GP_PIN(2, 9),
+				RCAR_GP_PIN(2, 10),	RCAR_GP_PIN(2, 11),
+				RCAR_GP_PIN(2, 12),	RCAR_GP_PIN(2, 13));
+DU_PFC_DAT8(du0_data8b,		DU0_DB0,		DU0_DB1,
+				DU0_DB2,		DU0_DB3,
+				DU0_DB4,		DU0_DB5,
+				DU0_DB6,		DU0_DB7);
+DU_PFC_PINS(du0_clk,		RCAR_GP_PIN(2, 15),	RCAR_GP_PIN(2, 16));
+DU_PFC_CLK(du0_clk,		DU0_DOTCLKO_UT0,	DU0_DOTCLKO_UT1);
+DU_PFC_PINS(du0_sync,		RCAR_GP_PIN(2, 17),	RCAR_GP_PIN(2, 18));
+DU_PFC_SYNC(du0_sync,		DU0_EXHSYNC_DU0_HSYNC,	DU0_EXVSYNC_DU0_VSYNC);
+DU_PFC_PINS(du0_disp,		RCAR_GP_PIN(2, 20));
+DU_PFC_DISP(du0_disp,		DU0_DISP);
 
 static const struct sh_pfc_pin_group pinmux_groups[] = {
 	SH_PFC_PIN_GROUP(audio_clk_a),
@@ -1893,6 +1931,12 @@ static const struct sh_pfc_pin_group pinmux_groups[] = {
 	SH_PFC_PIN_GROUP(vin1_data8),
 	SH_PFC_PIN_GROUP(vin1_clk),
 	SH_PFC_PIN_GROUP(vin1_sync),
+	SH_PFC_PIN_GROUP(du0_data8r),
+	SH_PFC_PIN_GROUP(du0_data8g),
+	SH_PFC_PIN_GROUP(du0_data8b),
+	SH_PFC_PIN_GROUP(du0_clk),
+	SH_PFC_PIN_GROUP(du0_sync),
+	SH_PFC_PIN_GROUP(du0_disp),
 };
 
 static const char * const audio_clk_groups[] = {
@@ -2093,6 +2137,15 @@ static const char * const vin1_groups[] = {
 	"vin1_sync",
 };
 
+static const char * const du0_groups[] = {
+	"du0_data8r",
+	"du0_data8g",
+	"du0_data8b",
+	"du0_clk",
+	"du0_sync",
+	"du0_disp",
+};
+
 static const struct sh_pfc_function pinmux_functions[] = {
 	SH_PFC_FUNCTION(audio_clk),
 	SH_PFC_FUNCTION(ether),
@@ -2120,6 +2173,7 @@ static const struct sh_pfc_function pinmux_functions[] = {
 	SH_PFC_FUNCTION(usb1),
 	SH_PFC_FUNCTION(vin0),
 	SH_PFC_FUNCTION(vin1),
+	SH_PFC_FUNCTION(du0),
 };
 
 static struct pinmux_cfg_reg pinmux_config_regs[] = {
-- 
1.8.2

