From 4c1895564220a770ca86c00b68b7205a865fe61c Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 7 Mar 2013 15:59:43 +0900
Subject: [PATCH 581/675] ARM: mach-shmobile: r8a7778: add GP2D clock support

---
 arch/arm/mach-shmobile/clock-r8a7778.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-shmobile/clock-r8a7778.c b/arch/arm/mach-shmobile/clock-r8a7778.c
index 2638021..4729800 100644
--- a/arch/arm/mach-shmobile/clock-r8a7778.c
+++ b/arch/arm/mach-shmobile/clock-r8a7778.c
@@ -118,7 +118,7 @@ enum {
 	MSTP331,
 	MSTP323, MSTP322, MSTP321,
 	MSTP311, MSTP310, MSTP309, MSTP308, MSTP307,
-	MSTP114, MSTP110, MSTP109, MSTP103, MSTP100,
+	MSTP114, MSTP110, MSTP109, MSTP107, MSTP103, MSTP100,
 	MSTP030,
 	MSTP029, MSTP028, MSTP027, MSTP026, MSTP025, MSTP024, MSTP023, MSTP022, MSTP021,
 	MSTP016, MSTP015, MSTP012, MSTP011, MSTP010, MSTP009, MSTP008,
@@ -138,6 +138,7 @@ static struct clk mstp_clks[MSTP_NR] = {
 	[MSTP114] = SH_CLK_MSTP32(&p_clk, MSTPCR1, 14, 0), /* Ether */
 	[MSTP110] = SH_CLK_MSTP32(&s_clk, MSTPCR1, 10, 0), /* VIN0 */
 	[MSTP109] = SH_CLK_MSTP32(&s_clk, MSTPCR1,  9, 0), /* VIN1 */
+	[MSTP107] = SH_CLK_MSTP32(&p_clk, MSTPCR1,  7, 0), /* GP2D */
 	[MSTP103] = SH_CLK_MSTP32(&s_clk, MSTPCR1,  3, 0), /* DU */
 	[MSTP100] = SH_CLK_MSTP32(&p_clk, MSTPCR1,  0, 0), /* USB0/1 */
 	[MSTP030] = SH_CLK_MSTP32(&p_clk, MSTPCR0, 30, 0), /* I2C0 */
@@ -177,6 +178,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_DEV_ID("sh-eth",	&mstp_clks[MSTP114]), /* Ether */
 	CLKDEV_DEV_ID("r8a7778-vin.0", &mstp_clks[MSTP110]), /* VIN0 */
 	CLKDEV_DEV_ID("r8a7778-vin.1", &mstp_clks[MSTP109]), /* VIN1 */
+	CLKDEV_DEV_ID("gp2d",		&mstp_clks[MSTP107]), /* GP2D */
 	CLKDEV_DEV_ID("rcar-du",	&mstp_clks[MSTP103]), /* DU */
 	CLKDEV_DEV_ID("ehci-platform", &mstp_clks[MSTP100]), /* USB EHCI port0/1 */
 	CLKDEV_DEV_ID("ohci-platform", &mstp_clks[MSTP100]), /* USB OHCI port0/1 */
-- 
1.8.2

