From cec95ed50a2f3977dfac429090c416a9b0cb3199 Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Tue, 11 Jun 2013 19:54:58 +0900
Subject: [PATCH 591/675] ARM: shmobile: r8a7778: add MERAM clock support

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/clock-r8a7778.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-shmobile/clock-r8a7778.c b/arch/arm/mach-shmobile/clock-r8a7778.c
index becfb3f..4a6b810 100644
--- a/arch/arm/mach-shmobile/clock-r8a7778.c
+++ b/arch/arm/mach-shmobile/clock-r8a7778.c
@@ -115,6 +115,7 @@ static struct clk *main_clks[] = {
 };
 
 enum {
+	MSTP513,
 	MSTP410,
 	MSTP408,
 	MSTP331,
@@ -128,6 +129,7 @@ enum {
 	MSTP_NR };
 
 static struct clk mstp_clks[MSTP_NR] = {
+	[MSTP513] = SH_CLK_MSTP32(&s3_clk, MSTPCR5, 13, 0), /* MERAM */
 	[MSTP410] = SH_CLK_MSTP32(&s3_clk, MSTPCR4, 10, 0), /* ICB */
 	[MSTP408] = SH_CLK_MSTP32(&s3_clk, MSTPCR4,  8, 0), /* VPC */
 	[MSTP331] = SH_CLK_MSTP32(&s4_clk, MSTPCR3, 31, 0), /* MMC */
@@ -175,6 +177,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_CON_ID("audio_clk_internal",	&s1_clk),
 
 	/* MSTP32 clocks */
+	CLKDEV_DEV_ID("uio_pdrv_genirq.2",	&mstp_clks[MSTP513]), /* MERAM */
 	CLKDEV_CON_ID("icb",		&mstp_clks[MSTP410]), /* ICB */
 	CLKDEV_DEV_ID("uio_pdrv_genirq.1",      &mstp_clks[MSTP408]), /* VPC */
 	CLKDEV_DEV_ID("sh_mmcif", &mstp_clks[MSTP331]), /* MMC */
-- 
1.8.2

