From a9062fd39749aa8cb9a913ae841fb905438616ba Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Mon, 22 Jul 2013 21:16:42 +0900
Subject: [PATCH 592/675] ARM: shmobile: r8a7778: add MERAM support

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7778.c | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/arch/arm/mach-shmobile/setup-r8a7778.c b/arch/arm/mach-shmobile/setup-r8a7778.c
index ddd2638..3be5c68 100644
--- a/arch/arm/mach-shmobile/setup-r8a7778.c
+++ b/arch/arm/mach-shmobile/setup-r8a7778.c
@@ -481,9 +481,42 @@ static struct platform_device vpc_device = {
 	.num_resources	= ARRAY_SIZE(vpc_resources),
 };
 
+/* MERAM */
+static struct uio_info meram_platform_data = {
+	.name		= "MERAM",
+	.version	= "0",
+	.irq		= -1,
+};
+
+static struct resource meram_resources[] = {
+	[0] = {
+		.name	= "MERAM",
+		.start	= 0xfd200000,
+		.end	= 0xfd200418 + 0x020 * 32 - 1,
+		.flags	= IORESOURCE_MEM,
+	},
+	[1] = {
+		.name	= "MERAM_MEM",
+		.start	= 0xfd480000,
+		.end	= 0xfd480000 + ((512 - 32) << 10) - 1, /* 32KiB reserved for cache */
+		.flags	= IORESOURCE_MEM,
+	}
+};
+
+static struct platform_device meram_device = {
+	.name		= "uio_pdrv_genirq",
+	.id		= 2,
+	.dev = {
+		.platform_data	= &meram_platform_data,
+	},
+	.resource	= meram_resources,
+	.num_resources	= ARRAY_SIZE(meram_resources),
+};
+ 
 static struct platform_device *r8a7778_devices[] __initdata = {
 	&ipmmu_device,
 	&vpc_device,
+	&meram_device,
 };
 
 void __init r8a7778_add_standard_devices(void)
-- 
1.8.2

