From 9f76b446df7ebdf385a095c4fed6c2e7802682dd Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 16 Apr 2013 14:51:34 +0900
Subject: [PATCH 600/675] ARM: shmobile: bockw: invoke SSB workaround

---
 arch/arm/mach-shmobile/board-bockw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index 19dbd4a..5ff4e00 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -477,6 +477,7 @@ static void __init bockw_init(void)
 	i2c_register_board_info(0, i2c0_devices,
 				ARRAY_SIZE(i2c0_devices));
 	r8a7778_meram_workaround();
+	r8a7778_ssb_workaround();
 
 	spi_register_board_info(spi_board_info,
 				ARRAY_SIZE(spi_board_info));
-- 
1.8.2

