From 5e76a5b74c403ad07f0c7fe97d0db6cdad357303 Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Fri, 14 Jun 2013 18:18:52 +0900
Subject: [PATCH 610/675] ARM: shmobile: bockw: Add suport mtd info

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/board-bockw.c | 48 ++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index ee5e9d5..b587983 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -22,6 +22,8 @@
 #include <linux/mfd/tmio.h>
 #include <linux/mmc/host.h>
 #include <linux/mtd/partitions.h>
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/physmap.h>
 #include <linux/pinctrl/machine.h>
 #include <linux/platform_device.h>
 #include <linux/platform_data/rcar-du.h>
@@ -176,6 +178,45 @@ static struct rcar_wdt_platform_config rcar_wdt_config = {
 	.timermode = WDT_WT_MODE,	/* Watchdog Mode */
 };
 
+/* MTD */
+static struct mtd_partition nor_flash_partitions[] = {
+	{
+		.name		= "loader",
+		.offset		= 0x00000000,
+		.size		= 2 * 1024 * 1024,
+		.mask_flags	= MTD_WRITEABLE,
+	},
+	{
+		.name		= "userdata",
+		.offset		= MTDPART_OFS_APPEND,
+		.size		= MTDPART_SIZ_FULL,
+	},
+};
+
+static struct physmap_flash_data nor_flash_data = {
+	.width		= 2,
+	.parts		= nor_flash_partitions,
+	.nr_parts	= ARRAY_SIZE(nor_flash_partitions),
+};
+
+static struct resource nor_flash_resources[] = {
+	[0]	= {
+		.start	= 0x00000000,
+		.end	= 0x04000000 - 1, /* 128MB(R0P0400C0001FS) */
+		.flags	= IORESOURCE_MEM,
+	}
+};
+
+static struct platform_device nor_flash_device = {
+	.name		= "physmap-flash",
+	.dev		= {
+		.platform_data	= &nor_flash_data,
+	},
+	.num_resources	= ARRAY_SIZE(nor_flash_resources),
+	.resource	= nor_flash_resources,
+};
+
+
 /* SSI/AK4554 */
 static struct rsnd_ssi_platform_info rsnd_ssi[] = {
 	RSND_SSI_UNUSED, /* SSI 0 */
@@ -455,6 +496,11 @@ static const struct pinctrl_map bockw_pinctrl_map[] = {
 				  "pwm6", "pwm6"),
 };
 
+static struct platform_device *bockw_devices[] __initdata = {
+	&nor_flash_device,
+};
+
+
 #define PFC	0xfffc0000
 #define PUPR4	0x110
 static void __init bockw_init(void)
@@ -548,6 +594,8 @@ static void __init bockw_init(void)
 	icb = clk_get(NULL, "icb");
 	if (!IS_ERR(icb))
 		clk_enable(icb);
+
+	platform_add_devices(bockw_devices, ARRAY_SIZE(bockw_devices));
 }
 
 static const char *bockw_boards_compat_dt[] __initdata = {
-- 
1.8.2

