From 5bda2efb13d208ee49d8b1ad62734725b21d7cd7 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 10 Sep 2013 13:28:10 +0900
Subject: [PATCH 1/4] hpb-dma: tidyup dummy channel

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 arch/arm/mach-shmobile/hpb-dmae.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-shmobile/hpb-dmae.c b/arch/arm/mach-shmobile/hpb-dmae.c
index 5e341cd..0a0ccb1 100644
--- a/arch/arm/mach-shmobile/hpb-dmae.c
+++ b/arch/arm/mach-shmobile/hpb-dmae.c
@@ -82,6 +82,8 @@ enum {
 */
 static const struct hpb_dmae_slave_config rcar_dmae_slaves[] = {
 	{
+		.id	= HPBDMA_DUMMY,
+	}, {
 		.id	= HPBDMA_SLAVE_SDHI0_TX,
 		.addr	= 0xffe4c000 + 0x30,
 		.dcr	= SPDS_16BIT | DMDL | DPDS_16BIT,
@@ -154,6 +156,8 @@ static const struct hpb_dmae_slave_config rcar_dmae_slaves[] = {
 
 /* comment out for not using Ch */
 static const struct hpb_dmae_channel rcar_dmae_channels[] = {
+	DMAE_CHANNEL(0, IRQ_DMAC_H(0), HPBDMA_DUMMY),
+
 	/* ch.21 SD0 */
 	DMAE_CHANNEL(0x540, IRQ_DMAC_H(21), HPBDMA_SLAVE_SDHI0_TX),
 	/* ch.22 SD0 */
-- 
1.7.9.5

