diff --git a/systems/fbdev/Makefile.am b/systems/fbdev/Makefile.am
index 2911ad8..ce268de 100644
--- a/systems/fbdev/Makefile.am
+++ b/systems/fbdev/Makefile.am
@@ -41,7 +43,8 @@ libdirectfb_fbdev_la_SOURCES = \
 libdirectfb_fbdev_la_LIBADD = \
 	$(top_builddir)/lib/direct/libdirect.la \
 	$(top_builddir)/lib/fusion/libfusion.la \
-	$(top_builddir)/src/libdirectfb.la
+	$(top_builddir)/src/libdirectfb.la \
+	-luiomux
 
 
 include $(top_srcdir)/rules/libobject.make
diff --git a/systems/fbdev/fbdev.c b/systems/fbdev/fbdev.c
index dd1c343..03e7361 100644
--- a/systems/fbdev/fbdev.c
+++ b/systems/fbdev/fbdev.c
@@ -93,6 +93,7 @@
 #include "agp.h"
 
 #include <core/core_system.h>
+#include <uiomux/uiomux.h>
 
 DFB_CORE_SYSTEM( fbdev )
 
@@ -517,6 +518,9 @@ system_initialize( CoreDFB *core, void **data )
           goto error;
      }
 
+     uiomux_register(dfb_fbdev->framebuffer_base, shared->fix.smem_start,
+          shared->fix.smem_len);
+
      if (ioctl( dfb_fbdev->fd, FBIOGET_VSCREENINFO, &shared->orig_var ) < 0) {
           D_PERROR( "DirectFB/FBDev: "
                     "Could not get variable screen information!\n" );
