Subject: Fix redefinition of mount_attr

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Hau Vo <hau.vo.ra@renesas.com>

---

diff --git a/include/lapi/fsmount.h b/include/lapi/fsmount.h
index 07eb42ffaf..5ae2a5f9e3 100644
--- a/include/lapi/fsmount.h
+++ b/include/lapi/fsmount.h
@@ -52,12 +52,6 @@
 /*
  * mount_setattr()
  */
-struct mount_attr {
-	uint64_t attr_set;
-	uint64_t attr_clr;
-	uint64_t propagation;
-	uint64_t userns_fd;
-};
 #endif
 
 #ifndef HAVE_FSOPEN
