From 1cdbf4ed4717eefa2b6237c1e63e351b62990522 Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@renesas.com>
Date: Fri, 23 Feb 2018 18:29:32 +0700
Subject: [PATCH] rpmsg: Add message to be able to configure RPMSG_VIRTIO from
 defconfig

Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
---
 drivers/rpmsg/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/rpmsg/Kconfig b/drivers/rpmsg/Kconfig
index 0fe6eac..5b1e61b 100644
--- a/drivers/rpmsg/Kconfig
+++ b/drivers/rpmsg/Kconfig
@@ -47,7 +47,7 @@ config RPMSG_QCOM_SMD
 	  platforms.
 
 config RPMSG_VIRTIO
-	tristate
+	tristate "Enable remote processor message and virtual I/O configure"
 	select RPMSG
 	select VIRTIO
 
-- 
1.9.1

