From a5fab3da14c65b8320bf77301f247ac13411ed9d Mon Sep 17 00:00:00 2001
From: Shunsuke Kataoka <shunsuke.kataoka.df@renesas.com>
Date: Fri, 26 Jul 2013 10:22:36 +0900
Subject: [PATCH 0751/1083] ARM: shmobile: r8a7790: Fix I2C resource size

Signed-off-by: Shunsuke Kataoka <shunsuke.kataoka.df@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 34f540e..8ab8147 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -987,7 +987,7 @@ static struct i2c_rcar_platform_data i2c_pd[] = {
 static struct resource rcar_i2c0_res[] = {
 	{
 		.start  = 0xe6508000,
-		.end    = (0xe6518000 - 1),
+		.end    = (0xe6510000 - 1),
 		.flags  = IORESOURCE_MEM,
 	}, {
 		.start  = gic_spi(287),
@@ -998,7 +998,7 @@ static struct resource rcar_i2c0_res[] = {
 static struct resource rcar_i2c1_res[] = {
 	{
 		.start  = 0xe6518000,
-		.end    = (0xe6528000 - 1),
+		.end    = (0xe6520000 - 1),
 		.flags  = IORESOURCE_MEM,
 	}, {
 		.start  = gic_spi(288),
@@ -1009,7 +1009,7 @@ static struct resource rcar_i2c1_res[] = {
 static struct resource rcar_i2c2_res[] = {
 	{
 		.start  = 0xe6530000,
-		.end    = (0xe6540000 - 1),
+		.end    = (0xe6538000 - 1),
 		.flags  = IORESOURCE_MEM,
 	}, {
 		.start  = gic_spi(286),
@@ -1020,7 +1020,7 @@ static struct resource rcar_i2c2_res[] = {
 static struct resource rcar_i2c3_res[] = {
 	{
 		.start  = 0xe6540000,
-		.end    = (0xe6550000 - 1),
+		.end    = (0xe6548000 - 1),
 		.flags  = IORESOURCE_MEM,
 	}, {
 		.start  = gic_spi(290),
-- 
1.8.3.2

