From ceb5c0df885a4ec066a9c446b07fa6137f2eddb1 Mon Sep 17 00:00:00 2001
From: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
Date: Mon, 29 Jul 2013 12:55:34 +0900
Subject: [PATCH 0765/1083] ARM: shmobile: r8a7790: Fix QSPI platform_device
 data

Signed-off-by: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 71cbaa8..3c7c1b6 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -842,9 +842,6 @@ static struct platform_device qspi_device = {
 	.id = 0,
 	.num_resources = ARRAY_SIZE(qspi_resources),
 	.resource	= qspi_resources,
-	.dev  = {
-		.platform_data = &spiflash_data,
-	},
 };
 
 static struct spi_board_info spi_info[] __initdata = {
-- 
1.8.3.2

