From f89e308e0aa828c7db0d2329488976a432e0eaf7 Mon Sep 17 00:00:00 2001
From: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
Date: Mon, 29 Jul 2013 13:40:10 +0900
Subject: [PATCH 0769/1083] ARM: shmobile: koelsch: Add MSIOF support

Signed-off-by: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
---
 arch/arm/mach-shmobile/board-koelsch.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-koelsch.c b/arch/arm/mach-shmobile/board-koelsch.c
index 9244987..834a379 100644
--- a/arch/arm/mach-shmobile/board-koelsch.c
+++ b/arch/arm/mach-shmobile/board-koelsch.c
@@ -31,11 +31,23 @@
 #include <linux/platform_data/rcar-du.h>
 #include <linux/platform_device.h>
 #include <linux/i2c.h>
+#include <linux/spi/spi.h>
 #include <mach/common.h>
 #include <mach/r8a7791.h>
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
 
+/* spidev for MSIOF */
+static struct spi_board_info spi_bus[] __initdata = {
+        {
+                .modalias       = "spidev",
+                .max_speed_hz   = 6000000,
+                .mode           = SPI_MODE_3,
+                .bus_num        = 1,
+                .chip_select    = 0,
+        },
+};
+
 static const struct pinctrl_map koelsch_pinctrl_map[] = {
 	/* SCIF0 (CN19: DEBUG SERIAL0) */
 	PIN_MAP_MUX_GROUP_DEFAULT("sh-sci.0", "pfc-r8a7791",
@@ -43,6 +55,14 @@ static const struct pinctrl_map koelsch_pinctrl_map[] = {
 	/* SCIF1 (CN20: DEBUG SERIAL1) */
 	PIN_MAP_MUX_GROUP_DEFAULT("sh-sci.1", "pfc-r8a7791",
 				  "scif1_data_d", "scif1"),
+
+	/* MSIOF0 */
+	PIN_MAP_MUX_GROUP_DEFAULT("spi_sh_msiof.0", "pfc-r8a7791",
+				  "msiof0_clk", "msiof0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("spi_sh_msiof.0", "pfc-r8a7791",
+				  "msiof0_ctrl", "msiof0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("spi_sh_msiof.0", "pfc-r8a7791",
+				  "msiof0_data", "msiof0"),
 };
 
 static void __init koelsch_add_standard_devices(void)
@@ -54,6 +74,9 @@ static void __init koelsch_add_standard_devices(void)
 	r8a7791_pinmux_init();
 
 	r8a7791_add_standard_devices();
+
+	/* spidev for MSIOF */
+	spi_register_board_info(spi_bus, ARRAY_SIZE(spi_bus));
 }
 
 static const char *koelsch_boards_compat_dt[] __initdata = {
-- 
1.8.3.2

