From b6056c8f751d072f55f8bab4fef7e6cea395d466 Mon Sep 17 00:00:00 2001
From: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
Date: Mon, 29 Jul 2013 13:49:29 +0900
Subject: [PATCH 0771/1083] ARM: shmobile: koelsch: Add QSPI support

Signed-off-by: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
---
 arch/arm/mach-shmobile/board-koelsch.c | 49 ++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-koelsch.c b/arch/arm/mach-shmobile/board-koelsch.c
index 834a379..0dd6a27 100644
--- a/arch/arm/mach-shmobile/board-koelsch.c
+++ b/arch/arm/mach-shmobile/board-koelsch.c
@@ -31,12 +31,58 @@
 #include <linux/platform_data/rcar-du.h>
 #include <linux/platform_device.h>
 #include <linux/i2c.h>
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/partitions.h>
 #include <linux/spi/spi.h>
+#include <linux/spi/flash.h>
 #include <mach/common.h>
 #include <mach/r8a7791.h>
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
 
+/* SPI Flash memory (Spansion S25FL512SAGMFIG11) */
+static struct mtd_partition spiflash_part[] = {
+	/* Reserved for user loader program, read-only */
+	[0] = {
+		.name = "loader_prg",
+		.offset = 0,
+		.size = SZ_256K,
+		.mask_flags = MTD_WRITEABLE,    /* read only */
+	},
+	/* Reserved for user program, read-only */
+	[1] = {
+		.name = "user_prg",
+		.offset = MTDPART_OFS_APPEND,
+		.size = SZ_4M,
+		.mask_flags = MTD_WRITEABLE,    /* read only */
+	},
+	/* All else is writable (e.g. JFFS2) */
+	[2] = {
+		.name = "flash_fs",
+		.offset = MTDPART_OFS_APPEND,
+		.size = MTDPART_SIZ_FULL,
+		.mask_flags = 0,
+	},
+};
+
+static struct flash_platform_data spiflash_data = {
+	.name		= "m25p80",
+	.parts		= spiflash_part,
+	.nr_parts	= ARRAY_SIZE(spiflash_part),
+	.type		= "s25fl512s",
+};
+
+static struct spi_board_info spi_info[] __initdata = {
+	{
+		.modalias	= "m25p80",
+		.platform_data	= &spiflash_data,
+		.mode		= SPI_MODE_0,
+		.max_speed_hz	= 30000000,
+		.bus_num	= 0,
+		.chip_select	= 0,
+	},
+};
+
 /* spidev for MSIOF */
 static struct spi_board_info spi_bus[] __initdata = {
         {
@@ -75,6 +121,9 @@ static void __init koelsch_add_standard_devices(void)
 
 	r8a7791_add_standard_devices();
 
+	/* QSPI flash memory */
+	spi_register_board_info(spi_info, ARRAY_SIZE(spi_info));
+
 	/* spidev for MSIOF */
 	spi_register_board_info(spi_bus, ARRAY_SIZE(spi_bus));
 }
-- 
1.8.3.2

