From 2554c8729c020c1402cd36ef4f1db23bb6ae0198 Mon Sep 17 00:00:00 2001
From: Nobuyuki HIRAI <nobuyuki.hirai.xe@renesas.com>
Date: Mon, 29 Jul 2013 20:15:40 +0900
Subject: [PATCH 0785/1083] ARM: shmobile: koelsch: Add VIN support

Signed-off-by: Nobuyuki HIRAI <nobuyuki.hirai.xe@renesas.com>
---
 arch/arm/mach-shmobile/board-koelsch.c | 100 +++++++++++++++++++++++++++++++++
 1 file changed, 100 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-koelsch.c b/arch/arm/mach-shmobile/board-koelsch.c
index bb14e0c..50eecdb 100644
--- a/arch/arm/mach-shmobile/board-koelsch.c
+++ b/arch/arm/mach-shmobile/board-koelsch.c
@@ -36,6 +36,7 @@
 #include <linux/spi/spi.h>
 #include <linux/spi/flash.h>
 #include <mach/common.h>
+#include <media/vin.h>
 #include <mach/r8a7791.h>
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
@@ -116,6 +117,102 @@ static const struct pinctrl_map koelsch_pinctrl_map[] = {
 				  "msiof0_ctrl", "msiof0"),
 	PIN_MAP_MUX_GROUP_DEFAULT("spi_sh_msiof.0", "pfc-r8a7791",
 				  "msiof0_data", "msiof0"),
+	/* VIN0 */
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_data_g", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_data_r", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_data_b", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_hsync_signal", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_vsync_signal", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_field_signal", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_data_enable", "vin0"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.0", "pfc-r8a7791",
+				  "vin0_clk", "vin0"),
+	/* VIN1 */
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.1", "pfc-r8a7791",
+				  "vin1_data", "vin1"),
+	PIN_MAP_MUX_GROUP_DEFAULT("vin.1", "pfc-r8a7791",
+				  "vin1_clk", "vin1"),
+};
+
+
+static struct i2c_board_info koelsch_i2c_camera[] = {
+	{ I2C_BOARD_INFO("adv7612", 0x4C), },
+	{ I2C_BOARD_INFO("adv7180", 0x20), },
+};
+
+static void camera_power_on(void)
+{
+	return;
+}
+
+static void camera_power_off(void)
+{
+	return;
+}
+
+static int adv7612_power(struct device *dev, int mode)
+{
+	if (mode)
+		camera_power_on();
+	else
+		camera_power_off();
+
+	return 0;
+}
+
+static int adv7180_power(struct device *dev, int mode)
+{
+	if (mode)
+		camera_power_on();
+	else
+		camera_power_off();
+
+	return 0;
+}
+
+static struct soc_camera_link adv7612_ch0_link = {
+	.bus_id = 0,
+	.power  = adv7612_power,
+	.board_info = &koelsch_i2c_camera[0],
+	.i2c_adapter_id = 2,
+	.module_name = "adv7612",
+};
+
+static struct soc_camera_link adv7180_ch1_link = {
+	.bus_id = 1,
+	.power  = adv7180_power,
+	.board_info = &koelsch_i2c_camera[1],
+	.i2c_adapter_id = 2,
+	.module_name = "adv7180",
+};
+
+static struct platform_device soc_camera_device[] = {
+	{
+		.name = "soc-camera-pdrv",
+		.id = 0,
+		.dev = {
+			.platform_data = &adv7612_ch0_link,
+		},
+	},
+	{
+		.name = "soc-camera-pdrv",
+		.id = 1,
+		.dev = {
+			 .platform_data = &adv7180_ch1_link,
+		},
+	},
+};
+
+static struct platform_device *koelsch_devices[] __initdata = {
+	&soc_camera_device[0],
+	&soc_camera_device[1],
 };
 
 static void __init koelsch_add_standard_devices(void)
@@ -128,6 +225,9 @@ static void __init koelsch_add_standard_devices(void)
 
 	r8a7791_add_standard_devices();
 
+	platform_add_devices(koelsch_devices,
+			     ARRAY_SIZE(koelsch_devices));
+
 	/* QSPI flash memory */
 	spi_register_board_info(spi_info, ARRAY_SIZE(spi_info));
 
-- 
1.8.3.2

