From f8bf292c0a085e5d28e32aac7a9e532b4fc0a983 Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Fri, 26 Jul 2013 09:45:59 +0900
Subject: [PATCH 0786/1083] ARM: shmobile: r8a7790: Add SCIF2 support

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 9aca575..0478162 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -157,7 +157,7 @@ void __init r8a7790_add_du_device(struct rcar_du_platform_data *pdata)
 }
 
 enum { SCIFA0, SCIFA1, SCIFB0, SCIFB1, SCIFB2, SCIFA2, SCIF0, SCIF1,
-       HSCIF0, HSCIF1 };
+       HSCIF0, HSCIF1, SCIF2 };
 
 static const struct plat_sci_port scif[] = {
 	SCIFA_DATA(SCIFA0, 0xe6c40000, gic_spi(144)), /* SCIFA0 */
@@ -170,6 +170,7 @@ static const struct plat_sci_port scif[] = {
 	SCIF_DATA(SCIF1, 0xe6e68000, gic_spi(153)), /* SCIF1 */
 	HSCIF_DATA(HSCIF0, 0xe62c0000, gic_spi(154)), /* HSCIF0 */
 	HSCIF_DATA(HSCIF1, 0xe62c8000, gic_spi(155)), /* HSCIF1 */
+	SCIF_DATA(SCIF2, 0xe6e56000, gic_spi(164)), /* SCIF2 */
 };
 
 static inline void r8a7790_register_scif(int idx)
-- 
1.8.3.2

