From 486bd12f9564069db4b5bb504fb773fdd524b99d Mon Sep 17 00:00:00 2001
From: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
Date: Tue, 30 Jul 2013 19:35:35 +0900
Subject: [PATCH 0798/1083] mmc: tmio_mmc: Fix copyright

Signed-off-by: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
---
 drivers/mmc/host/tmio_mmc_dma.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/tmio_mmc_dma.c b/drivers/mmc/host/tmio_mmc_dma.c
index 7f7ef93..ada18e9 100644
--- a/drivers/mmc/host/tmio_mmc_dma.c
+++ b/drivers/mmc/host/tmio_mmc_dma.c
@@ -1,6 +1,7 @@
 /*
  * linux/drivers/mmc/tmio_mmc_dma.c
  *
+ * Copyright (C) 2013 Renesas Electronics Corporation
  * Copyright (C) 2010-2011 Guennadi Liakhovetski
  *
  * This program is free software; you can redistribute it and/or modify
-- 
1.8.3.2

