From 8277b0716d9f708e02979df0294bbd519b2aac00 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 31 Jul 2013 15:45:17 +0900
Subject: [PATCH 1046/1083] drivers: Fix gpu order for i2c codec support

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/Makefile b/drivers/Makefile
index 1480802..05c451e 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -34,9 +34,6 @@ obj-$(CONFIG_REGULATOR)		+= regulator/
 obj-y				+= tty/
 obj-y				+= char/
 
-# gpu/ comes after char for AGP vs DRM startup
-obj-y				+= gpu/
-
 obj-$(CONFIG_CONNECTOR)		+= connector/
 
 # i810fb and intelfb depend on char/agp/
@@ -80,6 +77,8 @@ obj-$(CONFIG_INPUT)		+= input/
 obj-$(CONFIG_I2O)		+= message/
 obj-$(CONFIG_RTC_LIB)		+= rtc/
 obj-y				+= i2c/ media/
+# gpu/ comes after char for AGP vs DRM startup
+obj-y				+= gpu/
 obj-$(CONFIG_PPS)		+= pps/
 obj-$(CONFIG_PTP_1588_CLOCK)	+= ptp/
 obj-$(CONFIG_W1)		+= w1/
-- 
1.8.3.2

