From 3564d7a51e2451fcd2c747158f8e6532f618bb7b Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 31 Jul 2013 15:46:36 +0900
Subject: [PATCH 1047/1083] drm: Fix rcar-du order for i2c codec support

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/gpu/drm/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/Makefile b/drivers/gpu/drm/Makefile
index 539d345..c1ca0bf 100644
--- a/drivers/gpu/drm/Makefile
+++ b/drivers/gpu/drm/Makefile
@@ -44,5 +44,5 @@ obj-$(CONFIG_DRM_NOUVEAU) +=nouveau/
 obj-$(CONFIG_DRM_EXYNOS) +=exynos/
 obj-$(CONFIG_DRM_GMA500) += gma500/
 obj-$(CONFIG_DRM_UDL) += udl/
-obj-$(CONFIG_DRM_RCAR_DU) += rcar-du/
 obj-y			+= i2c/
+obj-$(CONFIG_DRM_RCAR_DU) += rcar-du/
-- 
1.8.3.2

