From db694abd6dc5a0710eb2e88f8d5e4963ddde8614 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 31 Jul 2013 15:51:49 +0900
Subject: [PATCH 1052/1083] drm: adv7511: Remove audio initialize

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/gpu/drm/i2c/adv7511_core.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i2c/adv7511_core.c b/drivers/gpu/drm/i2c/adv7511_core.c
index e687d9c..10bdf8b 100644
--- a/drivers/gpu/drm/i2c/adv7511_core.c
+++ b/drivers/gpu/drm/i2c/adv7511_core.c
@@ -1,5 +1,10 @@
 /**
- * Analog Devices ADV7511 HDMI transmitter driver
+ * drivers/gpu/drm/i2c/adv7511_core.c
+ *     This file is hdmi setting for adv7511.
+ *
+ * Copyright (C) 2013 Renesas Electronics Corporation
+ *
+ * * Analog Devices ADV7511 HDMI transmitter driver
  *
  * Copyright 2012 Analog Devices Inc.
  *
@@ -905,7 +910,6 @@ static int adv7511_probe(struct i2c_client *i2c,
 	adv7511->current_edid_segment = -1;
 
 	i2c_set_clientdata(i2c, adv7511);
-	adv7511_audio_init(&i2c->dev);
 
 	adv7511_set_link_config(adv7511, &link_config);
 
-- 
1.8.3.2

