From 6fadcc87316520922c41d84c46e6946122d2136c Mon Sep 17 00:00:00 2001
From: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
Date: Tue, 6 Aug 2013 17:21:30 +0900
Subject: [PATCH 1077/1083] mmc: sh_mobile_sdhi: Add disable auto CMD12 support

Signed-off-by: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
---
 drivers/mmc/host/sh_mobile_sdhi.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mmc/host/sh_mobile_sdhi.c b/drivers/mmc/host/sh_mobile_sdhi.c
index 8cff05d..c25e0e8 100644
--- a/drivers/mmc/host/sh_mobile_sdhi.c
+++ b/drivers/mmc/host/sh_mobile_sdhi.c
@@ -135,6 +135,15 @@ static bool sh_mobile_sdhi_dma_filter(struct dma_chan *chan, void *arg)
 	return true;
 }
 
+#define SH_MOBILE_SDHI_DISABLE_AUTO_CMD12	0x4000
+
+static void sh_mobile_sdhi_disable_auto_cmd12(int *val)
+{
+	*val |= SH_MOBILE_SDHI_DISABLE_AUTO_CMD12;
+
+	return ;
+}
+
 static void sh_mobile_sdhi_cd_wakeup(const struct platform_device *pdev)
 {
 	mmc_detect_change(dev_get_drvdata(&pdev->dev), msecs_to_jiffies(100));
@@ -182,6 +191,7 @@ static int __devinit sh_mobile_sdhi_probe(struct platform_device *pdev)
 	mmc_data->clk_enable = sh_mobile_sdhi_clk_enable;
 	mmc_data->clk_disable = sh_mobile_sdhi_clk_disable;
 	mmc_data->capabilities = MMC_CAP_MMC_HIGHSPEED;
+	mmc_data->disable_auto_cmd12 = sh_mobile_sdhi_disable_auto_cmd12;
 	if (p) {
 		mmc_data->flags = p->tmio_flags;
 		if (mmc_data->flags & TMIO_MMC_HAS_IDLE_WAIT)
-- 
1.8.3.2

