From 8f7226fa5f2b9aae1ebaf31031378e02dac60c32 Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Thu, 16 May 2013 15:01:31 +0900
Subject: [PATCH 08/14] net: Modify phy LED mode of sh_eth

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/net/sh_eth.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/net/sh_eth.c b/drivers/net/sh_eth.c
index 9788b75..68eb517 100644
--- a/drivers/net/sh_eth.c
+++ b/drivers/net/sh_eth.c
@@ -413,6 +413,12 @@ static int sh_eth_config(struct sh_eth_dev *eth, bd_t *bd)
 		goto err_phy_cfg;
 	}
 	phy = port_info->phydev;
+#if defined(CONFIG_ARCH_R8A7790)
+	ret = phy_read(phy, MDIO_DEVAD_NONE, 0x1e);
+	ret &= ~0xc000;
+	ret |= 0x4000;
+	phy_write(phy, MDIO_DEVAD_NONE, 0x1e, (u16)ret);
+#endif
 	ret = phy_startup(phy);
 	if (ret) {
 		printf(SHETHER_NAME ": phy startup failure\n");
-- 
1.7.3.4

