From 9947a9fbed7d8659bafd99cecda52b9574d5b8bb Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Mon, 29 Jul 2013 12:07:06 +0900
Subject: [PATCH 23/28] board: lager: Fix QoS setting

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
---
 board/renesas/lager/qos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/renesas/lager/qos.c b/board/renesas/lager/qos.c
index a12d5db..f2247de 100644
--- a/board/renesas/lager/qos.c
+++ b/board/renesas/lager/qos.c
@@ -63,7 +63,7 @@ void qos_init(void)
 
 	/* S3C -QoS */
 	s3c = (struct r8a7790_s3c *)S3C_BASE;
-	writel(0x80FF1D0D, &s3c->s3cadsplcr);
+	writel(0x80000000, &s3c->s3cadsplcr);
 	writel(0x0F0F0505, &s3c->s3crorr);
 	writel(0x0F0F0505, &s3c->s3cworr);
 
-- 
1.8.3.2

