From 9d88d2cb532858b02ead318ecb3fa89249f88557 Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Mon, 29 Jul 2013 13:28:59 +0900
Subject: [PATCH 27/28] board: koelsch: Add SPI Quad Access configuration

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
---
 include/configs/koelsch.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/koelsch.h b/include/configs/koelsch.h
index ff4571c..d3b89a0 100644
--- a/include/configs/koelsch.h
+++ b/include/configs/koelsch.h
@@ -145,6 +145,7 @@
 #define CONFIG_SH_QSPI
 #define CONFIG_SPI_FLASH
 #define CONFIG_SPI_FLASH_SPANSION
+#define CONFIG_SPI_FLASH_QUAD
 #define CONFIG_SYS_NO_FLASH
 #define CONFIG_SH_QSPI_BASE	0xE6B10000
 #endif
-- 
1.8.3.2

