From 5e7cf619598d5f67e099352808b5f86649459dca Mon Sep 17 00:00:00 2001
From: nhannguyen <you@example.com>
Date: Wed, 20 Jul 2016 10:14:38 +0700
Subject: [PATCH] Edit dma_buf_export with new prototype

---
 drv/mmngr_buf_drv.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drv/mmngr_buf_drv.c b/drv/mmngr_buf_drv.c
index 37e8e70..732e40c 100755
--- a/drv/mmngr_buf_drv.c
+++ b/drv/mmngr_buf_drv.c
@@ -285,12 +285,18 @@ static const struct dma_buf_ops dmabuf_ops = {
 static int mm_ioc_export_start(int __user *arg, struct MM_BUF_PRIVATE *priv)
 {
 	struct MM_BUF_PARAM	tmp;
+    DEFINE_DMA_BUF_EXPORT_INFO(exp_info);
 
 	if (copy_from_user(&tmp, arg, sizeof(struct MM_BUF_PARAM)))
 		goto exit;
 
-	priv->dma_buf = dma_buf_export(priv, &dmabuf_ops, tmp.size, O_RDWR);
-	if (IS_ERR(priv->dma_buf))
+
+	exp_info.priv = priv;
+    exp_info.ops = &dmabuf_ops;
+    exp_info.size = tmp.size;
+    exp_info.flags = O_RDWR;
+    priv->dma_buf = dma_buf_export(&exp_info);
+    if (IS_ERR(priv->dma_buf))
 		goto exit;
 
 	tmp.buf = dma_buf_fd(priv->dma_buf, 0);
-- 
1.9.1

