diff --git a/vspm_if-module/files/vspm_if/drv/vspm_if_main.c b/vspm_if-module/files/vspm_if/drv/vspm_if_main.c
index f47645f..0b785f8 100644
--- a/vspm_if-module/files/vspm_if/drv/vspm_if_main.c
+++ b/vspm_if-module/files/vspm_if/drv/vspm_if_main.c
@@ -64,6 +64,7 @@
 #include <linux/dma-mapping.h>
 #include <linux/fs.h>
 #include <linux/ioctl.h>
+#include <linux/of_address.h>
 
 #include "vspm_public.h"
 #include "vspm_if.h"
