From d8844580854ebb48f72ea4043789fe3db1b258de Mon Sep 17 00:00:00 2001
From: Tomohito Esaki <etom@igel.co.jp>
Date: Fri, 12 Apr 2013 18:24:23 +0900
Subject: [PATCH 18/32] armadillo800eva Add clock entry for SGX540

Signed-off-by: Damian Hobson-Garcia <dhobsong@igel.co.jp>
---
 arch/arm/mach-shmobile/clock-r8a7740.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-shmobile/clock-r8a7740.c b/arch/arm/mach-shmobile/clock-r8a7740.c
index 069b9f9..67519efd 100644
--- a/arch/arm/mach-shmobile/clock-r8a7740.c
+++ b/arch/arm/mach-shmobile/clock-r8a7740.c
@@ -460,6 +460,7 @@ static struct clk div6_clks[DIV6_NR] = {
 enum {
 	MSTP128, MSTP127, MSTP125,
 	MSTP116, MSTP111, MSTP100, MSTP117, MSTP101, MSTP107, MSTP103,
+	MSTP112,
 
 	MSTP230,
 	MSTP222,
@@ -482,6 +483,7 @@ static struct clk mstp_clks[MSTP_NR] = {
 	[MSTP117] = SH_CLK_MSTP32(&div4_clks[DIV4_B],	SMSTPCR1, 17, 0), /* LCDC1 */
 	[MSTP116] = SH_CLK_MSTP32(&div6_clks[DIV6_SUB],	SMSTPCR1, 16, 0), /* IIC0 */
 	[MSTP111] = SH_CLK_MSTP32(&div6_clks[DIV6_SUB],	SMSTPCR1, 11, 0), /* TMU1 */
+	[MSTP112] = SH_CLK_MSTP32(&div6_clks[DIV4_ZG],	SMSTPCR1, 12, 0), /* SGX */
 	[MSTP100] = SH_CLK_MSTP32(&div4_clks[DIV4_B],	SMSTPCR1,  0, 0), /* LCDC0 */
 	[MSTP101] = SH_CLK_MSTP32(&div4_clks[DIV4_B],	SMSTPCR1,  1, 0), /* VPU5HA2 */
 	[MSTP103] = SH_CLK_MSTP32(&div4_clks[DIV4_B],	SMSTPCR1,  3, 0), /* VIO6C */
@@ -556,6 +558,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_CON_ID("sub_clk",		&div6_clks[DIV6_SUB]),
 
 	/* MSTP32 clocks */
+	CLKDEV_CON_ID("sgx",			&mstp_clks[MSTP112]),
 	CLKDEV_DEV_ID("sh_mobile_lcdc_fb.0",	&mstp_clks[MSTP100]),
 	CLKDEV_DEV_ID("uio_dmem_genirq.0",	&mstp_clks[MSTP101]),
 	CLKDEV_DEV_ID("uio_pdrv_genirq.1",	&mstp_clks[MSTP103]),
-- 
1.8.2

