From 901068e83bb465dd3cb968933ed89696548e077e Mon Sep 17 00:00:00 2001
From: Hiroyuki Yokoyama <hiroyuki.yokoyama.vx@renesas.com>
Date: Fri, 21 Aug 2015 16:44:42 +0900
Subject: arm: rcar_gen3: Add skip of relocation

Signed-off-by: Hiroyuki Yokoyama <hiroyuki.yokoyama.vx@renesas.com>
---
 arch/arm/lib/crt0_64.S | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/lib/crt0_64.S b/arch/arm/lib/crt0_64.S
index 1654011..ef90b04 100644
--- a/arch/arm/lib/crt0_64.S
+++ b/arch/arm/lib/crt0_64.S
@@ -87,6 +87,7 @@ zero_gd:
 	ldr	x18, [x18, #GD_BD]		/* x18 <- gd->bd */
 	sub	x18, x18, #GD_SIZE		/* new GD is below bd */
 
+#ifndef CONFIG_RCAR_GEN3
 	adr	lr, relocation_return
 	ldr	x9, [x18, #GD_RELOC_OFF]	/* x9 <- gd->reloc_off */
 	add	lr, lr, x9	/* new return address after relocation */
@@ -94,6 +95,7 @@ zero_gd:
 	b	relocate_code
 
 relocation_return:
+#endif
 
 /*
  * Set up final (full) environment
-- 
1.8.3.2

